src/bulls_and_cows_lib/game.cpp

```diff
@@ -9,52 +9,118 @@
-               "       Display the number of remaining possible codes so far\n"
-               "       Wait for 2 seconds\n"
-               "       Pick a random attempt among in the list of remaining possible codes\n"
-               "       Compare the computer's attempt with the secret code and deduce the number of bulls and cows\n"
-               "       Add the computer's attempt to the list of attempts of the board\n"
-               "       Remove all the codes that are incompatible with the attempt's feedback from the list of "
-               "possible codes\n"
-               "    WHILE not end of game\n"
-               "    Display the board and the list of attempts so far\n"
-               "    Display a message telling if the computer won or lost\n";
+        //bulls_and_cows::generate_all_possible_codes(game_options);
+
+
+        PossibleSolutions myavc = bulls_and_cows::generate_all_possible_codes(game_options);
+        std::cout << "Solution generated with " << myavc.codes.size() << " solutions\n";
+        Board myavcboard = bulls_and_cows::create_board(game_options);
+
+        while (!bulls_and_cows::is_end_of_game(game_options, myavcboard) && !bulls_and_cows::is_win(game_options, myavcboard))
+        {
+            // First display of the empty board
+            bulls_and_cows::display_board(std::cout, game_options, myavcboard);
+            std::cout << "Solutions still are " << myavc.codes.size() << " solutions\n";
+
+            // Asking attemp the computer to choose a random possibilitie
+            Code avctemp = bulls_and_cows::pick_random_attempt(myavc);
+            std::cout << "Choosen code is " << avctemp.value << "\n";
+
+            AttemptAndFeedback newattemp;
+            newattemp.attempt = avctemp;
+            newattemp.feedback = bulls_and_cows::compare_attempt_with_secret_code(avctemp, myavcboard.secret_code);
+            //std::cout << "Bulls : " << newattemp.feedback.bulls << " Cows : " << newattemp.feedback.cows<<"\n";
+            myavcboard.attempts_and_feedbacks.push_back(newattemp);
+            std::cout << "Secret code is : " << myavcboard.secret_code.value << "\n";
+            if (bulls_and_cows::is_win(game_options, myavcboard))break;
+            bulls_and_cows::remove_incompatible_codes_from_possible_solutions(newattemp, myavc);
+            Sleep(2000);
+        }
     }
 
-    void configure_game_options(GameOptions& game_options)
+    void configure_game_options(GameOptions & game_options)
     {
-        std::cout << "TODO:\n"
-                     "    DO\n"
-                     "       Display the current game options\n"
-                     "       Display the game options menu\n"
-                     "       Ask the user to type its choice\n"
-                     "       Treat the user's choice\n"
-                     "    UNTIL user's choice is to go back to main menu\n";
+        std::ifstream in{"game_options.txt"};
```

Comment:
Ce n'est pas la partie 3, mais pour info, il faut éviter de créer un stream de fichier tant qu'on n'en a pas besoin, car ça ouvre le fichier immediatement, mais ne le ferme que quand la variable in est détruite (en sortie du bloc de la fonction).

Reply:
cf prochain comment